gen_random_array.h

```diff
@@ -40,4 +40,22 @@ std::array<T, len> random_array(T minimum = std::numeric_limits<T>::min(), T max
     return std::move(out);
 }
 
+// Use vector for arrays too big to live on the stack
+template<typename T, size_t len>
+requires int_for_distrib<T> || std::floating_point<T>
+std::vector<T> random_array_v(T minimum = std::numeric_limits<T>::min(), T maximum = std::numeric_limits<T>::max())
+{
+    std::mt19937 gen{std::random_device{}()};
+    typename DistType<T>::type dist(minimum, maximum);
+
+    std::vector<T> out(len);
+
+    for(size_t i = 0; i < out.size(); ++i)
+    {
+        out[i] = dist(gen);
+    }
+
+    return std::move(out);
```

Comment:
Do we actually need to call `std::move` here or does the compiler do that (or something better) on its own?

Reply:
I think that's right but there is some gap in my compiler knowledge here so I'm not totally sure if it's guaranteed or what. I admit I'm not totally sure why you need `std::move` to exist if the compiler always (at any level of optimization?) does this for you.